src/actions/session.ts

```diff
@@ -43,8 +45,13 @@ export const setTokens = ({
     refreshToken
   });
 
-export const setUser = (user: { name: string; role: Role; grade: number; story: Story }) =>
-  action(actionTypes.SET_USER, user);
+export const setUser = (user: {
+  name: string;
+  role: Role;
+  grade: number;
+  story?: Story;
```

Comment:
Does this ever get updated? e.g. When the student completes the main quest of the mission, story should be updated to the next story so that they will see the new script (if available) when they next log in.

If not, we can try and implement it using external actions at the end of each story's main quest. (in a separate PR)

Reply:
> Does this ever get updated? e.g. When the student completes the main quest of the mission, story should be updated to the next story so that they will see the new script (if available) when they next log in.

It should be updated.

Reply:
> It should be updated.

As in it should already be done? Or it should be done in future?

Reply:
As in, how the story works is that when the assignment is finished. It will then update to the latest story after the user logs back in. Because it always gets the earliest unopen story.